src/main/java/com/gimp/GimPlugin.java

```diff
@@ -341,6 +355,71 @@ else if (
 			updateGhostMode(config.ghostMode());
 		}
 	}
+	@Subscribe
+	public void onMenuOptionClicked(MenuOptionClicked event)
+	{
+		if (!client.isKeyPressed(KeyCode.KC_SHIFT) || client.isMenuOpen() || group.getGimps().isEmpty() || !config.pings())
+		{
+			return;
+		}
+
+		Tile selectedSceneTile = client.getSelectedSceneTile();
+		if (selectedSceneTile == null)
+		{
+			return;
+		}
+
+		boolean isOnCanvas = false;
+
+		for (MenuEntry menuEntry : client.getMenuEntries())
+		{
+			if (menuEntry == null)
+			{
+				continue;
+			}
+
+			if ("walk here".equalsIgnoreCase(menuEntry.getOption()))
+			{
+				isOnCanvas = true;
+			}
+		}
+
+		if (!isOnCanvas)
+		{
+			return;
+		}
+
+		event.consume();
+		final TilePing tilePing = new TilePing(selectedSceneTile.getWorldLocation());
+		onTilePing(tilePing);
```

Comment:
If we move this below line 395, where we set the member ID, it will show the correct gimp color locally (on your screen when you ping). Right now it just uses the default color of red because the `tilePing` `memberId` has not yet been set equal to the gimp's UUID.